chore: fix CI as dependencies no longer support Go 1.20

blgm opened the pull request:
There was a previous strategy to test with the version of Go specified in the go.mod file. However some dependencies no longer support this version, so we are moving to testing with only supported versions of Go.

Hopefully this change will unblock Dependabot PRs #150 and #152

moleske reviewed (approved):
seems alright.  non blocking question/clarification -

Nothing in the change set seems to be about go 1.23.  Is that on purpose?  The go.mod sets the go version to 1.22

blgm commented:
Thanks @moleske. The problem started when Go 1.23 came out and Ginkgo and staticcheck appear to have dropped support for running with unsupported versions of Go. So you are correct that it's not really the Go version that's caused this, but changes in the dependencies, and I've updated the PR title to reflect this. I updated the go.mod file to Go 1.22 as this is the current lowest supported version, and we know that there are potential problems running with older versions, so we might as well force folks to use Go 1.22 as a minimum rather than have them wasting time with debugging any issues.
